worker/src/data.ts

```diff
@@ -74,5 +93,22 @@ export const createQueueData = (): QueueData => ({
   count_users: [],
 });
 
+export const generateMetadata = (
+  payload: SanitizedPayload,
+  timestamp: number,
+  metadata?: Metadata | null
+): Metadata => ({
+  created: metadata ? metadata.created : timestamp,
+  updated: timestamp,
+  installation_type: payload.installation_type,
+  version: payload.version,
+  country: payload.country,
+  extra: Object.keys(payload).some((key) => BasePayloadKeys.includes(key)),
+});
+
 export const bumpValue = (current?: number): number =>
   !current ? 1 : current + 1;
+
+export function isMetadata(metadata: unknown): metadata is Metadata {
```

Comment:
You don't actually verify `metadata is Metadata` ? You just test if `extra` is defined or not. I think that this needs a better name ?